02 - HTML intro/form.html

```diff
@@ -18,26 +18,29 @@
                 </p>
                 <p>
                     <label for="tel">Phone number:</label>
-                    <input id="tel" type="text"pattern="[\+]\d{2}[\-]\d{3}[\-]\d{3}[\-]\d{3}" >
+                    <input id="tel" type="text" pattern="[\+]\d{2}[\-]\d{3}[\-]\d{3}[\-]\d{3}" required >
                 </p>
                 <p>
                     <label for="mail">E-mail:</label>
-                    <input id="mail" type="email" pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$">
+                    <input id="mail" type="email" pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$" required>
                 </p>
                 <p>
-                    <label for="date" pattern="[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|1[0-9]|2[0-9]|3[01])
-">Date:</label>
-                    <input id="date" type="date">
+                    <label for="date">Date:</label>
+                    <input id="date" type="date" pattern="[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|1[0-9]|2[0-9]|3[01])
```

Comment:
@gorzat Attribute pattern is only allowed when the input type is email, password, search, tel, text, or url. Work around here would be to change this field to text type and use pattern on it.